pgx_async.opam

```diff
@@ -11,7 +11,7 @@ install: [jbuilder install]
 remove: [jbuilder uninstall]
 
 depends: [
-  "async" {= "v0.9.1"}
```

Comment:
Is this change relevant to circle ci?

Reply:
This is required because we're using `opam install --deps-only pgx_async`, and `v0.9.1` doesn't exist. I moved the change to a different commit with a description.